e2e/testgames/game/Dockerfile

```
@@ -17,12 +17,42 @@ RUN cd e2e/testgames/game && go build -v -o /go/bin/app
+
+# assert seems like it will be least likely to change
+COPY assert assert
+
+# we assume here that testgames will rarely change
+COPY e2e/testgames e2e/testgames
+
+# assumes we are most often testing changes to cardinal code
+COPY cardinal cardinal
+
+# cardinal and assert are part of the go mods, so this must be done after they are copied
+RUN cd e2e/testgames && go mod download
```

Hadolint

[warning] Use WORKDIR to switch to a directory

(DL3003)